jukepoot/puppet_parser.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python
+"""A parser of /var/log/puppet.log.
+
+Imports argparse for parsing arguments and then the script
+accepts file path and year as optional argument. 
+"""
+
+
+import argparse 
+from datetime import datetime
+import dateutil.parser
+import re
+import collections
+import os
+
+
+def main():
+	parser = argparse.ArgumentParser()
+	parser.add_argument(help="parse the given filename", type=str, dest="filename", action="store")
+	parser.add_argument("-y","--year", help="parse file with given year", type=int, dest="year", action="store")
+	args = parser.parse_args()
+	if args.year:
+		print("year to use: " + str(args.year))
```

Comment:
no need to typecast to str... rather ues
`
print("year to use: {0:d}".format(args.year))`